cloudpickle/cloudpickle.py

```diff
@@ -141,8 +143,22 @@ def _whichmodule(obj, name):
     return None
 
 
+if sys.version_info[:2] < (3, 7):
+    def _get_module_attr(obj):
+        if isinstance(obj, typing.TypeVar):
+            return None
+        return getattr(obj, '__module__', None)
+else:
+    def _get_module_attr(obj):
+        return getattr(obj, '__module__', None)
+
+
 def _is_global(obj, name=None):
     """Determine if obj can be pickled as attribute of a file-backed module"""
+    return _get_module_if_global(obj, name)
+
+
+def _get_module_if_global(obj, name=None):
```

Comment:
May I suggest to rename this to `_lookup_module_and_qualname`?

Reply:
Also this helper function is not trivial which makes me think that we should have dedicated unittest for it to provide a harness for future refactoring and smaller maintenance operations on the cloudpickle code base.

Could you please at least add a couple of unittest to call this helper `_get_module_if_global` on typevars as part of this PR and later in a  subsequent PR we will add more unittests to also covers regular, locally defined, interactively defined classes, methods, functions and lambdas.

For this PR we should test:

- locally defined typevar,
- `mypkg.T` (custom module from site-packages)
- `typing.AnyStr` (standard library module)

This test will have to introspect `sys.version_info` because the expected result in not the same for different versions of Python.

Reply:
Done. The behaviour is the same in all supported versions.